### PR TITLE
Fixing the N=1 subspace expansion

### PR DESCRIPTION
**Pull request by LHerviou**
Here is the fix.

The code can probably be optimised, but it should be perfectly working now.
I needed it to benchmark vumps_mpo anyway.

Sorry about the random useless commits (the previous one). Apparently, the rebase was not done automatically.
I will try to do that before the next PR.

### CODE REVIEW

**Comment by LHerviou** [comment]
Playing around with it, I think the error estimate might be off.

File: test/test_vumps.jl [hunk 1]
```diff
@@ -38,15 +38,15 @@ using Random
...
-      # Flux density is inconsistent for odd unit cells
-      isodd(nsites) && continue
-    end
+    #if conserve_qns
+    # Flux density is inconsistent for odd unit cells
+    #  isodd(nsites) && continue
+    #end
 
-    space_ = fill(space_shifted(model, 0; conserve_qns=conserve_qns), nsites)
+    space_ = fill(space_shifted(model, 1; conserve_qns=conserve_qns), nsites)
```
Comment (mtfishman) [inline test/test_vumps.jl:49]: Shouldn't it be `0` for even-site unit cells and `1` for odd-site unit cells?

**Comment by mtfishman** [comment]
It seems like the code could be shared/compressed between the `N = 1` and `N > 1` cases, but I'm ok with dealing with that later if you don't have time to look into that right now. I think it is more important for this to be working so we can test it out.

File: test/test_vumps.jl (hunk 1 above)
Comment (LHerviou) [reply test/test_vumps.jl:49]: Due to the 2-modularity, for the chosen initial state, anything goes when the number of sites is even..

**Comment by LHerviou** [comment]
I agree, there is definitely some overlap. I just wanted a quick and dirty fix to be able to work on the vumps_mpo.

Building on what you did in vumps_localham, I have something that appears to be working for nsites(\psi ) = 1

File: test/test_vumps.jl (hunk 1 above)
Comment (mtfishman) [reply test/test_vumps.jl:49]: Ah I see, thanks for the explanation.

**Comment by mtfishman** [comment]
So should we go ahead and merge this?

**Comment by LHerviou** [comment]
I think so.
Please let me know if any other problems appear.

Once it is done and I manage to clean the history, I will do a PR for the MPO ref of the InfiniteTensorSum, and then hopefully of the MatrixMPO representation.

**Comment by mtfishman** [comment]
Great, thanks!

**Pull request merged**